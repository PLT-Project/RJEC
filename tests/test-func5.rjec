struct dummy {
    x int;
    y int;
    z bool;
    t char;
}

func foo (x int, y int, z struct dummy) struct dummy {
    var i struct dummy;
    i = struct dummy {
        x : x,
        y : y,
        z : z.z,
        t : z.t
    };
    z = struct dummy {
        x : 5
    };
    return i;
}

func main () {
    var i, j struct dummy;
    j = struct dummy {
        x : 10,
        t : 'z'
    };
    i = foo(2, 3, j);

    printi(i.x);
    printi(i.y);
    printb(i.z);
    printc(i.t);

    printi(j.x);
}