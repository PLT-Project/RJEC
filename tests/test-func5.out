2
3
false
z
10